include/exadg/operators/generalized_laplace_operator.h

```diff
@@ -0,0 +1,849 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_OPERATORS_GENERALIZED_LAPLACE_OPERATOR_H_
+#define INCLUDE_EXADG_OPERATORS_GENERALIZED_LAPLACE_OPERATOR_H_
+
+#include <exadg/functions_and_boundary_conditions/evaluate_functions.h>
+#include <exadg/grid/grid_utilities.h>
+#include <exadg/operators/interior_penalty_parameter.h>
+#include <exadg/operators/operator_base.h>
+#include <exadg/operators/variable_coefficients.h>
+#include <exadg/poisson/user_interface/boundary_descriptor.h>
+#include <variant>
+
+namespace ExaDG
+{
+namespace GeneralizedLaplace
+{
+/**
+ * Free function for a generic expression of the multiplication with coefficients of different
+ * ranks.
+ */
+template<typename T, int coefficient_rank, int dim, typename Number>
+static inline DEAL_II_ALWAYS_INLINE //
+  T
+  coeff_mult(dealii::Tensor<coefficient_rank, dim, Number> const & coeff, T const & x)
+{
+  if constexpr(coefficient_rank == 4)
+    return dealii::double_contract<2, 0, 3, 1>(coeff, x);
+  else
+    return coeff * x;
+}
```

Comment:
"Multiplication **of what** with coefficients of different ranks". The word "different" is misunderstandable. I would write "... from the left with a coefficient tensor of rank `coefficient rank`".

In my opinion, the variable names `T`, `coeff_mult`, `x` are too short. Also `2,0,3,1` is somewhat hard to understand.

Reply:
I'll change the sentence, rename the variables and put a comment about the double contraction👍

Reply:
Let's also make the function clearer by using full names.